scipy/stats/stats.py

```diff
@@ -3800,18 +3821,24 @@ def alexandergovern(*args):
     return AlexanderGovernResult(A, p)
 
 
-def _alexandergovern_input_validation(args):
+def _alexandergovern_input_validation(args, nan_policy):
     if len(args) < 2:
         raise TypeError(f"2 or more inputs required, got {len(args)}")
 
-    # the current intended behavior is that the samples are flattened into 1d
+    # input arrays are flattened
     args = [np.ravel(np.asarray(arg, dtype=float)) for arg in args]
 
-    for arg in args:
+    for i, arg in enumerate(args):
         if np.size(arg) <= 1:
-            raise ValueError("Input samples cannot be zero-dimensional.")
-        if False in np.isfinite(arg):
+            raise ValueError("Input sample size must be greater than one.")
+        if arg.ndim != 1:
+            raise ValueError("Input samples must be one-dimensional")
```

Comment:
When I commented before about this check, it was before I noticed that args are traveled. 
```suggestion
```
Please check me, but I don't think you can get this error since you're raveling all the args first.

Reply:
I agree. I couldn't think of an input that would cause this type of situation besides ndim being 0, but that case seems to be covered by input size.

If you think any of these cases: `arg = `
- `2`
-  `[2]`
- `[]`

would be better explained by the one dimensional error than size error, I could add more complex logic.

Reply:
No, I think since you are raveling whatever is input, it really is a size error and not a dimension-related error; the function doesn't care about the dimensions.